packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -262,6 +262,20 @@ class ScaffoldMessengerState extends State<ScaffoldMessenger> with TickerProvide
   ///
   /// ** See code in examples/api/lib/material/scaffold/scaffold_messenger_state.show_snack_bar.0.dart **
   /// {@end-tool}
+  ///
+  /// ## Relative positioning of floating SnackBars
+  ///
+  /// A [SnackBar] with [SnackBar.behavior] set to [SnackBarBehavior.floating] might
+  /// not be visible when the widget provided to [Scaffold.floatingActionButton]
+  /// is taking up entire space. As a result [SnackBar] is presented off the visible area.
+  /// Consider using constrained Widget's size in [Scaffold.floatingActionButton].
```

Comment:
There is actually more than one widget in Scaffold that affects this positioning in addition to the floating action button. 
Scaffold.persistentFooterButtons, Scaffold.bottomNavigationBar both affect it as well.

How about something along the lines of...

```suggestion
  /// ## Relative positioning of floating SnackBars
  ///
  /// A [SnackBar] with [SnackBar.behavior] set to [SnackBarBehavior.floating] is
  /// positioned above the widgets provided to [Scaffold.floatingActionButton],
  /// [Scaffold.persistentFooterButtons], and [Scaffold.bottomNavigationBar].
  /// If some or all of these widgets take up enough space such that the SnackBar
  /// would not be visible when positioned above them, an error will be thrown.
  /// In this case, consider constraining the size of these widgets to allow room for
  /// the SnackBar to be visible.
```